compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -3567,13 +3567,16 @@ void OMR::ValuePropagation::transformArrayCloneCall(TR::TreeTop *callTree, OMR::
    if (method->getRecognizedMethod() == TR::java_lang_J9VMInternals_primitiveClone)
       objNode = callNode->getLastChild();
 
-   if (!performTransformation(comp(), "%sInlining array clone call [%p] as new array and arraycopy\n", OPT_DETAILS, callNode))
-      return;
-
    TR_OpaqueClassBlock *j9arrayClass = cloneInfo->_clazz;
-   bool isFixedClass = cloneInfo->_isFixed;
-
    TR_OpaqueClassBlock *j9class = comp()->fe()->getComponentClassFromArrayClass(j9arrayClass);
+   bool isPrimitiveClass = TR::Compiler->cls.isPrimitiveClass(comp(), j9class);
+
+   if (isPrimitiveClass ? !cg()->getSupportsPrimitiveArrayCopy()
+                        : !cg()->getSupportsReferenceArrayCopy())
+      return;
```

Comment:
This needs a diagnostic to report that we are giving up because the codegen doesn't support the transformation.

Reply:
Added `traceMsg()`.